### PR TITLE
[generator][routing] Ignore "highway=construction".

### PR DESCRIPTION
**Pull request by vng**
Closes https://github.com/organicmaps/organicmaps/issues/3929
Closes https://github.com/organicmaps/organicmaps/issues/7257

Can check after new data generation.

### CODE REVIEW

File: generator/routing_index_generator.cpp [hunk 1]
```diff
@@ -68,28 +69,35 @@ class VehicleMaskBuilder final
...
 
   VehicleMask CalcOneWayMask(FeatureType & f) const
   {
-    return CalcMask(f, [&](VehicleModelInterface const & model, FeatureType & f) {
-      return model.IsOneWay(f);
+    feature::TypesHolder const types(f);
+    return CalcMask([&](VehicleModelInterface const & model)
+    {
+      return model.IsOneWay(types);
     });
```
Comment (biodranik) [inline generator/routing_index_generator.cpp:88]: The current syntax looks strange, maybe use this one?
```suggestion
        {
          return model.IsOneWay(types);
        });
```

**Review by biodranik — APPROVED** [review]

File: generator/routing_index_generator.cpp (hunk 1 above)
Comment (vng) [reply generator/routing_index_generator.cpp:88]: Well, I used to write lambdas like this and I think it is comfortable. When lambda's function is +1 offset from main code. No need to make any _strange_ alignment and put 3-4 offsets:
```
prologue;
foo([&]()
{
  // lambda code with 1 offset
});
epilogue;
```
vs
```
prologue;
foo([&]()
       {
         // lambda code with 5 offsets
       });
epilogue;

```

**Comment by pastk** [comment]
Will explicit `bicycle/foot=yes` override this behavior? I.e. will it build a foot route over a constructed highway in such a case?
Ref: https://wiki.openstreetmap.org/wiki/Tag:highway=construction#How_to_map_the_start_of_a_road_repair_or_reconstruction

**Comment by vng** [comment]
No, and that was the main reason for this bugfix. highway=construction + bicycle=yes or foot=yes should **not** present in a route. Mentioned issues have exact these examples.

**Comment by pastk** [comment]
Then either the wiki is wrong or those samples are mismapped..

**Pull request merged**